### PR TITLE
chore: fix underlaying patch library [PHX-2907]

### PR DESCRIPTION
**Pull request by marcolink**
Switch the underlaying library to generate JSON Patch objects for entities from different environments.

### CODE REVIEW

File: test/unit/engine/utils/create-patch.test.ts [hunk 1]
```diff
@@ -9,8 +9,8 @@ describe('createPatch', () => {
     const targetEntry = targetEntriesFixture.items[0] as unknown as Entry<any>
     const patch = createPatch({ targetEntry, sourceEntry })
     expect(patch).to.deep.equal([
-      { op: 'replace', path: '/fields/slug/en-US', value: 'home-page' },
       { op: 'replace', path: '/fields/title/en-US', value: 'Home page' },
+      { op: 'replace', path: '/fields/slug/en-US', value: 'home-page' },
```
Comment (marcolink) [inline test/unit/engine/utils/create-patch.test.ts:13]: The order of creation changed with the new library.

File: src/engine/utils/create-patch.ts [hunk 2]
```diff
@@ -1,17 +1,13 @@
...
+import { generateJSONPatch } from 'generate-json-patch'
 
 export const createPatch = ({ targetEntry, sourceEntry }: { targetEntry: Entry<any>; sourceEntry: Entry<any> }) => {
-  return format(entryDiff.diff(targetEntry, sourceEntry))
+  // Temp fix until the types from generate-json-patch are fixed
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  return generateJSONPatch(targetEntry, sourceEntry, {
+    propertyFilter: function (name: string) {
+      return !['sys'].includes(name)
```
Comment (andreascful) [inline src/engine/utils/create-patch.ts:10]: In changeset-create we had the filter like this:
```
const { length } = pathInfo(context.path)
      // ignore sys prop on root level
      if (length === 1) {
        return !['sys'].includes(name)
      }
      return true
```

Would you say that would be safer here?

Comment (marcolink) [reply src/engine/utils/create-patch.ts]: Great catch @andreascful 🙏

File: test/unit/engine/utils/create-patch.test.ts [hunk 3]
```diff
@@ -9,15 +9,20 @@ describe('createPatch', () => {
...
     ])
   })
   it('creates a patch for added list items', () => {
     const sourceEntry = sourceEntriesFixture.items[2] as unknown as Entry<any>
     const targetEntry = targetEntriesFixture.items[2] as unknown as Entry<any>
     const patch = createPatch({ targetEntry, sourceEntry })
     expect(patch).to.deep.equal([
+      {
```
Comment (andreascful) [inline test/unit/engine/utils/create-patch.test.ts:21]: Super happy that the usage of the new library even caught this faulty test. 👍

**Review by andreascful — APPROVED** [review]

File: test/unit/engine/utils/create-patch.test.ts [hunk 4]
```diff
@@ -9,15 +9,20 @@ describe('createPatch', () => {
...
+        op: 'replace',
+        path: '/fields/modules/en-US/4/sys/id',
+        value: '6gFiJvssqQ62CMYqECOu2M',
+      },
```
Comment (marcolink) [inline test/unit/engine/utils/create-patch.test.ts:25]: By introducing the path check for `sys` only on root, we noticed that the old library didn't catch the nested id change within an array object. This is actually a known problem we already saw in other places with `jsondiffpatch`.

File: src/engine/utils/create-patch.ts [hunk 5]
```diff
@@ -1,17 +1,18 @@
...
 export const createPatch = ({ targetEntry, sourceEntry }: { targetEntry: Entry<any>; sourceEntry: Entry<any> }) => {
-  return format(entryDiff.diff(targetEntry, sourceEntry))
+  // Temp fix until the types from generate-json-patch are fixed
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  return generateJSONPatch(targetEntry, sourceEntry, {
+    propertyFilter: function (name: string, context) {
+      const { length } = pathInfo(context.path)
+      // ignore sys prop on root level
+      if (length === 1) {
```
Comment (ruderngespra) [inline src/engine/utils/create-patch.ts:12]: should we also ignore metadata here?

Comment (marcolink) [reply src/engine/utils/create-patch.ts:12]: @ruderngespra good point. 

Ignoreing it now - also added a test for it!

**Review by andreascful — APPROVED** [review]

File: src/engine/utils/create-patch.ts (hunk 5 above)
Comment (andreascful) [reply src/engine/utils/create-patch.ts:12]: I'd personally advise against it. We already exclude metadata at an early step in the pipeline so it should never get this far. I am not sure if creating more places where we ignore it makes it easier to understand and change it a later point.

Comment (marcolink) [reply src/engine/utils/create-patch.ts:12]: That's a valid point @andreascful - please feel free to revert it with another PR as this is already merged 🙏

**Pull request merged**